test/core/util/test_config.cc

```diff
@@ -90,11 +92,32 @@ gpr_timespec grpc_timeout_milliseconds_to_deadline(int64_t time_ms) {
           GPR_TIMESPAN));
 }
 
-void grpc_test_init(int /*argc*/, char** argv) {
+namespace {
+void grpc_test_init_parse_argv(int* argc, char** argv) {
+  // flags to look for and consume
+  constexpr char poller_flag[] = "--poller=";
+  constexpr size_t poller_flag_len = sizeof(poller_flag) - 1;
+  for (int i = 1; argc && i < *argc; ++i) {
+    if (strncmp(poller_flag, argv[i], poller_flag_len) == 0) {
+      gpr_setenv("GRPC_POLL_STRATEGY", argv[i] + poller_flag_len);
+      // remove the spent argv
+      for (int j = i; j < *argc - 1; ++j) {
+        argv[j] = argv[j + 1];
+      }
+      --i;
```

Comment:
Instead of incrementing in the loop statement and decrementing here, suggest incrementing only if we didn't consume the argument.

Reply:
Done